x/kavamint/types/params.go

```diff
@@ -56,29 +54,34 @@ func (p *Params) Validate() error {
 	return validateStakingRewardsApy(p.StakingRewardsApy)
 }
 
+// String implements fmt.Stringer
+func (p Params) String() string {
+	return fmt.Sprintf(`Params:
+	CommunityPoolInflation: %s
+	StakingRewardsApy: %s`,
+		p.CommunityPoolInflation, p.StakingRewardsApy)
+}
+
 func validateCommunityPoolInflation(i interface{}) error {
 	rate, ok := i.(sdk.Dec)
 	if !ok {
 		return fmt.Errorf("invalid parameter type: %T", i)
 	}
-	return validateRateWithinBounds(rate)
+	return validateRate(rate)
 }
 
 func validateStakingRewardsApy(i interface{}) error {
 	rate, ok := i.(sdk.Dec)
 	if !ok {
 		return fmt.Errorf("invalid parameter type: %T", i)
 	}
-	return validateRateWithinBounds(rate)
+	return validateRate(rate)
 }
 
-// validateRateWithinBounds ensure that the given rate falls within the allowed bounds: [0, MaxMintingRate]
-func validateRateWithinBounds(rate sdk.Dec) error {
-	if rate.BigInt().Sign() == -1 {
-		return fmt.Errorf("rate must be >= 0")
-	}
-	if MaxMintingRate.LT(rate) {
-		return fmt.Errorf("rate out of bounds. the max allowed rate is %s", MaxMintingRate)
+// validateRate ensures rate is properly initialized (non-nil), not negative, and not greater than the max rate
+func validateRate(rate sdk.Dec) error {
+	if rate.IsNil() || rate.IsNegative() || rate.GT(MaxMintingRate) {
```

Comment:
It's important to check that sdk.Dec values are not nill as they will panic if somehow set